src/domains/agents/analyzer/analyzer_agent.py

```
@@ -166,8 +166,8 @@ async def learn(self, beliefs: List[Belief], intentions: List[Intention], contex
         """Learn from analysis results"""
         try:
             for belief in beliefs:
-                if belief.confidence > self.min_confidence_threshold:
-                    if belief.type == "automation_opportunity":
+                if belief.confidence > self.min_confidence_threshold and belief.type == BeliefType.KNOWLEDGE:
+                    if belief.type == BeliefType.KNOWLEDGE:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)